src/librbd/cache/rwl/LogEntry.h

```diff
@@ -159,6 +159,35 @@ class WriteLogEntry : public GenericWriteLogEntry {
                                   const WriteLogEntry &entry);
 };
 
+class DiscardLogEntry : public GenericWriteLogEntry {
+public:
+  DiscardLogEntry(std::shared_ptr<SyncPointLogEntry> sync_point_entry,
+                  const uint64_t image_offset_bytes, const uint64_t write_bytes)
+    : GenericWriteLogEntry(sync_point_entry, image_offset_bytes, write_bytes) {
+    ram_entry.discard = 1;
+  };
+  DiscardLogEntry(const uint64_t image_offset_bytes, const uint64_t write_bytes)
+    : GenericWriteLogEntry(nullptr, image_offset_bytes, write_bytes) {
+    ram_entry.discard = 1;
+  };
+  DiscardLogEntry(const DiscardLogEntry&) = delete;
+  DiscardLogEntry &operator=(const DiscardLogEntry&) = delete;
+  inline unsigned int write_bytes() override {
+    /* The valid bytes in this ops data buffer. */
+    return 0;
+  };
+  inline unsigned int bytes_dirty() override {
```

Comment:
Nit: (fix the const-ness of the virtual method)
```suggestion
  unsigned int bytes_dirty() const override {
```

Reply:
Done